src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1175 @@ const ActorInit En_Kakasi_InitVars = {
+            this->unk190++;
+            return;
+        case 1:
+            // end of twirl, start shaking back and forth
+            if (this->unk204 == 0 && this->animeFrameCount <= currentFrame) {
+                this->unk204 = 0x14;
+                this->unk190++;
+                EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SIDEWAYS_SHAKING);
+            }
+            return;
+        case 2:
+            // camera view on the left
+            if (this->unk204 == 0) {
+                this->unk204 = 0x14;
+                this->unk190++;
+            }
+            return;
+        case 3:
+            // camera view on the right
+            if (this->unk204 == 0) {
+                this->unk204 = 0x14;
+                this->unk190++;
+            }
+            return;
+        case 4:
+            // camera view close up front
+            if (this->unk204 == 0) {
+                this->unk204 = 0x14;
+                this->unk190++;
+            }
+            return;
+        case 5:
+            // camera view zoomed out front
+            if (this->unk204 == 0) {
+                this->unk204 = 0xF;
+                this->unk190++;
+            }
+            return;
+        case 6:
+            // zoom in and sweep down in from case 5
+            if (this->unk204 == 0) {
+                this->unk204 = 0xF;
+                this->unk190++;
+            }
+            return;
+        case 7:
+            // zoom in close from case 6
+            if (this->unk204 == 0) {
+                this->unk204 = 0xA;
+                this->unk190++;
+            }
+            return;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_8096F8D8.s")
+        case 8:
+        case 9:
+        case 10:
+        case 11:
+        case 12:
+        case 13:
+            // multiple quick angle cuts viewing from multiple sides
+            if (this->unk204 == 0) {
+                this->unk190++;
+                this->unk204 = 0xA;
+                if (this->unk190 == 0xE) {
+                    func_800B7298(globalCtx, &this->actor, 0x49);
+                    func_80165690();
+                    this->unk204 = 0x14;
+                }
+            }
+            return;
+        case 14:
+            // goes off once for some camera changes,
+            // otherwise its the end when camera is back to normal and link is confused
+            Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 1000, 0);
+            if (this->unk204 == 0) {
+                player = PLAYER;
+                func_80169DCC(globalCtx, 0, Entrance_CreateIndexFromSpawn(0), player->unk_3CE, 0xBFF, &player->unk_3C0,
+                              player->unk_3CC);
+                func_80169EFC(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_8096FA18.s")
+                if (0) {}
+                if (gSaveContext.time >= 0xC001 || gSaveContext.time < 0x4000) {
```

Comment:
Don't forget we have `tools/timeconv.py`
```suggestion
                if (gSaveContext.time > CLOCK_TIME(18, 0) || gSaveContext.time < CLOCK_TIME(6, 0)) {
```